pkg/ec2provider/ec2provider.go

```diff
@@ -0,0 +1,274 @@
+package ec2provider
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/aws-iam-authenticator/pkg/httputil"
+	"sync"
+	"time"
+)
+
+const (
+	// max limit of k8s nodes support
+	max_channel_size = 5000
```

Comment:
Nit: This could be set higher, then it won't have to be changed later on.

Reply:
How about 10K ?